cmake/modules/Builduring.cmake

```diff
@@ -0,0 +1,33 @@
+function(build_uring)
+  include(ExternalProject)
+  set(IO_URING_SRC "${CMAKE_BINARY_DIR}/src/liburing/src")
+  set(IO_URING_INCLUDE "${IO_URING_SRC}/include")
+  set(IO_URING_LIB "${IO_URING_SRC}")
+
+  find_program (MAKE_EXECUTABLE NAMES make gmake)
+  if(CMAKE_MAKE_PROGRAM MATCHES "make")
+    set(make_cmd "$(MAKE)")
+  else()
+    set(make_cmd "${MAKE_EXECUTABLE}")
+  endif()
+
+  ExternalProject_Add(liburing_ext
+    DOWNLOAD_DIR ${CMAKE_BINARY_DIR}/src/
+    GIT_REPOSITORY http://git.kernel.dk/liburing
```

Comment:
Within FreeBSD fetching sources whilest building is seriously frowned up.
And in Ceph most external sources are also fetch upfront, thru submodules.
Can that be done for this code as well?

Reply:
@wjwithagen 
developers could decide to build out the library from source code or use the system installed library. This PR could support it.
If the developers decide to use the system installed library, it's better not to clone the added submodule in ceph/.git/config since it won't be used.

BTW,  some of the external sources also use this method to do build.
```
ceph/cmake/modules$ grep GIT_REPOSITORY ./ -rn
./BuildFIO.cmake:11:    GIT_REPOSITORY "https://github.com/axboe/fio.git"
./Buildpmem.cmake:18:      GIT_REPOSITORY "https://github.com/ceph/pmdk.git"
./AddCephTest.cmake:33:      GIT_REPOSITORY "https://github.com/google/gtest-parallel.git"
```

Reply:
@changchengx 
This was the PR I was asked to review, and thus the comment.
The good thing of having all these at the submodules is that it is centralized, easy to find, and easy to maintain.
I would comment the same was I asked to review any of the other subprojects.

Reply:
@wjwithagen 
How to solve the problem if the developer wants to use OS installed liburing instead of building it from source code?

Reply:
@wjwithagen i don't think io uring is a feature supported by FreeBSD.

Reply:
@tchaikov 
No ATM it is not, I've read about a Dev that was trying to implemnt it. But never heart from it after that.

I made it a comment, and not a `code change` because it was more a general comment.
I like the submodules feature, as it centralizes the list of external included sources.
To be honest I have a local patch I use to remove unwanted submodules so that they do not get fetched. Preventing larges downloads.

Then again is dashboard the largest exemption, and that really does case a lot of extra work on FreeBSD.